x-pack/plugins/ml/public/application/datavisualizer/file_based/file_datavisualizer.tsx

```diff
@@ -40,60 +39,62 @@ export const FileDataVisualizerPage: FC = () => {
 
   const [FileDataVisualizer, setFileDataVisualizer] = useState<FileDataVisualizerSpec | null>(null);
 
-  const links: ResultLink[] = useMemo(
+  const asyncLinkCards = useMemo(
```

Comment:
These show up when accessing the file view from within the ML plugin, but not when I upload a file from the Kibana Home page - Upload a file.

Reply:
If i'm remembering correctly, we purposely didn't add any ML links to the file data vis when being used inside kibana's Add Data page.
It would be a lot more work to register the links on start up. We'd have to add a registry for the links to be stored in. Whereas currently the links are passed on component render.

Also most users will not be jumping from there to ML.